django_react_proj/users-fe/src/components/Home/Home.js

```diff
@@ -1,19 +1,36 @@
 import React, { useEffect, useState } from 'react';
-import { Col, Container, Row } from 'reactstrap';
+import { Button, Modal, Col, Container, Row } from 'reactstrap';
 import axios from 'axios';
+import Calendar from 'react-calendar';
 import styles from './Home.module.css';
 import UserList from '../UserList/UserList';
 import NewUserModal from '../NewUserModal/NewUserModal';
 import { API_URL } from '../../constants';
 import Time from '../RealTime/RealTime';
+import 'react-calendar/dist/Calendar.css';
+
+const getUserForView = (users = [], date = new Date()) => {
+  const m = (date.getMonth() + 1).toString().padStart(2, '0');
+  const y = date.getFullYear();
+  const d = date.getDate().toString().padStart(2, '0');
+  return users
+    .filter((user) => user.registrationDate === `${y}-${m}-${d}`)
+    .sort((userA, userB) => userA.deadline.localeCompare(userB.deadline));
+};
+const getPreviousDay = (date) => new Date(date.getFullYear(), date.getMonth(), date.getDate() - 1);
+const getNextDay = (date) => new Date(date.getFullYear(), date.getMonth(), date.getDate() + 1);
 
 const Home = () => {
   const [users, setUsers] = useState();
-
+  const [currentDate, setCurrentDate] = useState(new Date());
+  const [modalFlag, setModalFlag] = useState(false);
+  const toggle = () => {
+    setModalFlag((previous) => !previous);
+  };
+  const button = <Button onClick={toggle}>Kalendarz</Button>;
   const getUsers = () => {
     axios.get(API_URL).then((res) => setUsers(res.data));
```

Comment:
what about using `async/await`? And maybe try to catch some errors if request fail

Reply:
It has been fixed on another branch